### ISSUE TITLE
Provide ECDH interface

### ISSUE THREAD

**Issue by tomato42** [warner/python-ecdsa#128]
While it's possible to fairly easily use existing code for ECDH key agreement, it's not really documented and handling of the shared secret form is tricky.

Provide classes for handling private key shares, public key shares and deriving shared secret. Reuse code from VerifyingKey and SigningKey.

**Comment by merlokk** [comment]
Ill add PR
```
def ecdh(curve, PrivateKey, PublicKey):

    pub = ecdsa.VerifyingKey.from_string(PublicKey[1:], curve=curve, hashfunc=sha256)
    prv = ecdsa.SigningKey.from_string(PrivateKey, curve=curve, hashfunc=sha256)

    # shared secret = PUBtheirs * PRIVATEours
    result = pub.pubkey.point * prv.privkey.secret_multiplier

    res = ecdsa.ecdsa.int_to_string(result.x())
    delta = curve.baselen - len(res)
    if delta > 0:
        res = b"\x00" * delta + res
    return res
```

**Comment by tomato42** [comment]
@merlokk I was more thinking along the lines of a separate class that makes it easier to handle ECDH in general.

What I had in mind was 4 operations:
 1. Selection of the curve for ECDH
 2. Ingestion of the shared key of the peer
 3. Generation of our shared key
 4. Derivation of the shared secret

(With 2 and 3 allowed to happen in either order, probably allowing the selection of curve with ingestion of either will be nice to have for DER/PEM files)

The differences from `VerifyingKey` and `SigningKey` I had in mind were:
 * lack of our shared key verification (if the ephemeral mode is used)
   * the peer's shared key still needs to be validated
 * no need to specify curve for `from_der`, `from_pem` and `from_string` methods for digestion of shared key (for `from_der` and `from_prem` code should still read the curve from the file, but abort if it doesn't match the expected one)
 * verification if the derived secret is safe (not zero or `INFINITY`)

We can use `openssl pkeyutl -derive -inkey key.pem -peerkey pubkey.pem -out secret` for interoperability testing.

**Comment by merlokk** [comment]
I have added basic part into #149
and add a class in next PR

**Comment by tomato42** [comment]
fixed by #149
